xformers/components/__init__.py

```diff
@@ -29,7 +29,7 @@
 
 
 def build_multi_head_attention(
-    multi_head_config: Union[MultiHeadDispatchConfig, Dict[str, Any]],
+    multi_head_config: Dict[str, Any],
```

Comment:
Didn't find any existing use case of calling this with `MultiHeadDispatchConfig` and having this hard dependency here on a specific type of dispatch complicates matters a little bit.

Reply:
We can do this
```
if isinstance(multi_head_config, MultiHeadDispatchConfig):
    multi_head_config = dataclasses.asdict(multi_head_config)
else:
    multi_head_config = multi_head_config.copy()
```